libs/ui/src/printed_ballot.test.tsx

```diff
@@ -0,0 +1,163 @@
+  // intact
+  return {
+    ...original,
+    randomBase64: () => 'CHhgYxfN5GeqnK8KaVOt1w',
+  };
+});
+
+interface TestCase {
+  description: string;
+  electionDefinition: ElectionDefinition;
+  ballotStyleId: string;
+  precinctId: string;
+  votes: {
+    [key: string]: string | string[] | Candidate;
+  };
+  isLiveMode?: boolean;
+}
+
+const testCases: TestCase[] = [
+  {
+    description:
+      'PrintedBallot renders votes for candidate contests and yes-no contests',
+    electionDefinition: electionSampleDefinition,
+    ballotStyleId: '5',
+    precinctId: '21',
+    votes: {
+      president: 'barchi-hallaren',
+      'lieutenant-governor': 'norberg',
+      'question-a': ['yes'],
+      'question-b': ['no'],
+    },
+  },
+  {
+    description:
+      'PrintedBallot renders votes for MS either-neither contests (both either-neither and pick-one vote provided)',
+    electionDefinition: electionWithMsEitherNeitherDefinition,
+    ballotStyleId: '1',
+    precinctId: '6525',
+    votes: {
+      '750000015': ['yes'],
+      '750000016': ['yes'],
+    },
+  },
+  {
+    description:
+      'PrintedBallot renders votes for MS either-neither contests (only either-neither vote provided)',
+    electionDefinition: electionWithMsEitherNeitherDefinition,
+    ballotStyleId: '1',
+    precinctId: '6525',
+    votes: {
+      '750000015': ['no'],
+    },
+  },
+  {
+    description:
+      'PrintedBallot renders votes for MS either-neither contests (only pick-one vote provided)',
+    electionDefinition: electionWithMsEitherNeitherDefinition,
+    ballotStyleId: '1',
+    precinctId: '6525',
+    votes: {
+      '750000016': ['no'],
+    },
+  },
+  {
+    description: 'PrintedBallot renders when no votes',
+    electionDefinition: electionWithMsEitherNeitherDefinition,
+    ballotStyleId: '1',
+    precinctId: '6525',
+    votes: {},
+  },
+  {
+    description: 'PrintedBallot renders when no votes and in live mode',
+    electionDefinition: electionWithMsEitherNeitherDefinition,
+    ballotStyleId: '1',
+    precinctId: '6525',
+    votes: {},
+    isLiveMode: true,
+  },
+  {
+    description: 'PrintedBallot renders votes for write-in candidates',
+    electionDefinition: electionWithMsEitherNeitherDefinition,
+    ballotStyleId: '1',
+    precinctId: '6525',
+    votes: {
+      '775020876': {
+        id: 'write-in__HOT DOG',
+        isWriteIn: true,
+        name: 'HOT DOG',
+      },
+    },
+  },
+  {
+    description:
+      'PrintedBallot renders remaining choices for multi-seat contests',
+    electionDefinition: electionSampleDefinition,
+    ballotStyleId: '12',
+    precinctId: '23',
+    votes: {
+      'city-council': ['eagle', 'smith'],
+    },
+  },
+  {
+    description: 'PrintedBallot renders when no seal',
+    electionDefinition: electionSampleNoSealDefinition,
+    ballotStyleId: '5',
+    precinctId: '21',
+    votes: {},
+  },
+];
```

Comment:
When I read these snapshot tests, I don't really feel like I can understand well what they are testing. The snapshots are too large and too obscure. Is it possible to write explicit test assertions to help show what differences we're looking for in each case? (That may be out of scope for this change, but if you did get a good sense of it while working here, this might be a good time to document it via tests)

Reply:
Mm really good point. Lemme make some tweaks!

Reply:
Taken care of in https://github.com/votingworks/vxsuite/pull/1736/commits/3323d27858ed67879280884a6a1b0cabc9d2ff45